core/templates/components/summary-tile/story-summary-tile.component.ts

```diff
@@ -0,0 +1,182 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for a canonical story tile.
+ */
+
+import { Component, OnInit } from '@angular/core';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { TopicViewerDomainConstants } from 'domain/topic_viewer/topic-viewer-domain.constants';
+import { Input } from '@angular/core';
+import { UrlService } from 'services/contextual/url.service';
+import { WindowDimensionsService } from 'services/contextual/window-dimensions.service';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { AssetsBackendApiService } from 'services/assets-backend-api.service';
+import { AppConstants } from 'app.constants';
+import { StorySummary } from 'domain/story/story-summary.model';
+
+@Component({
+  selector: 'oppia-story-summary-tile',
+  templateUrl: 'story-summary-tile.component.html'
+})
+export class StorySummaryTileComponent implements OnInit {
+  @Input() classroomUrlFragment: string;
+  @Input() storySummary: StorySummary;
+  @Input() topicUrlFragment: string;
+  initialCount: number;
+  chaptersDisplayed: number;
+  nodeCount: number;
+  completedStoriesCount: number;
+  storyProgress: number;
+  thumbnailUrl: string;
+  showButton: boolean;
+  constructor(
+    private urlInterpolationService: UrlInterpolationService,
+    private urlService: UrlService,
+    private windowDimensionsService: WindowDimensionsService,
+    private assetsBackendApiService: AssetsBackendApiService
+  ) {}
+  circumference = (20 * 2 * Math.PI);
+  gapLength = 5;
+  EXPLORE_PAGE_PREFIX = '/explore/';
```

Comment:
move these above the constructor and add a blank line to separate members and the constructor.

Reply:
done